### ISSUE TITLE
错误于UseMethod("train"): "train"没有适用于"formula"目标对象的方法

### ISSUE THREAD

**Issue by tjhwangxiong** [l-magnificence/Mime#91]
```
library(Mime1)
load("./Example.ici.Rdata")
load("./genelist.Rdata")
res.ici <- ML.Dev.Pred.Category.Sig(train_data = list_train_vali_Data$training,
                                    list_train_vali_Data = list_train_vali_Data,
                                    candidate_genes = genelist,
                                    methods = c('nb','svmRadialWeights','rf',
                                                'kknn','adaboost','LogitBoost',
                                                'cancerclass'),
                                    seed = 5201314,
                                    cores_for_parallel = 60
)

```
Error:
---loading the function---
[1] "There existing 42 genes in candidate genes"
[1] "Intersetion of the candidate genes and the colnames of the provided data"
[1] "There existing 42 genes in candidate genes, colnames of training data, colnames of validation data"
**错误于UseMethod("train"): "train"没有适用于"formula"目标对象的方法**

**Comment by ventonchen** [comment]
> Have your environment loaded the package "generics" which is conflict with "caret" ?

Nowadays I can't load caret without generics. When I run res.ici <- ML.Dev.Pred.Category.Sig() after I installed genetics and library(caret), the same error "Error in UseMethod("train") : 
  no applicable method for 'train' applied to an object of class "formula"
" come up again, how can I solve it

**Comment by xiaochenggushiduo** [comment]
> Have your environment loaded the package "generics" which is conflict with "caret" ?

Hello, if generics are loaded, how should it be handled?

**Comment by huangy100** [comment]
the method :
assign(
  "train",
  caret::train,
  envir = .GlobalEnv
)
res.ici <- ML.Dev.Pred.Category.Sig(
  train_data = list_train_vali_Data$training,
  list_train_vali_Data = list_train_vali_Data,
  candidate_genes = genelist,
  methods = c('nb','svmRadialWeights','rf','kknn','adaboost','LogitBoost','cancerclass'),
  seed = 5201314,
  cores_for_parallel = 30
)


the core cause:Attaching package: ‘generics’
The following object is masked from ‘package:caret’:
    train

generics::train() mathematical function will cover caret::train() function because  generics package  is loaded   accompanied  with    BiocGenerics package  .
